### ISSUE TITLE
Update the diagrams of the Algebraic / Topological structures

### ISSUE THREAD

**Issue by tirix** [metamath/set.mm#1904]
I volunteer to update the [diagrams of the Algebraic / Topological structures](http://us.metamath.org/mpeuni/mmtopstr.html) in Metamath.
I think some new structures have been defined and may be added to the diagrams:
- new topological structures: Magma (Mgm), Semigroups (SGrp), maybe the non-unital Ring (Rng)
- new graph definitions "as structures": (UHGraph , USHGraph, UPGraph, UMGraph, USPGraph, USGraph, FinUSGraph, ComplGraph, ComplUSGraph)
- categories: Cat, SetCat, CatCat, others?
- possibly, geometries: TarskiG & co.

Is it worth doing the update? Am I missing any? Any suggestion?

**Comment by nmegill** [comment]
> Is it worth doing the update?

Yes.  Stephen Wolfram was impressed when I pointed him to your page:  "That's nice.  I tried to construct things like this around 25 years ago; this is better."

**Comment by savask** [comment]
I also think that an update is very much welcome, and I have a suggestion: maybe algebraic and topological diagrams could have links to corresponding structures in them? Right now links are provided only in the algebraic-topological hierarchy.

**Comment by tirix** [comment]
Thanks for your (very gratifying) feedbacks!

I've updated the first diagram (algebraic hierarchy) and added the links in PR #1907 .

I've tried to provide links even to the arrows. I'm keeping this issue open for the last 3 bullets.

**Comment by avekens** [comment]
Sorry for my late remarks...

I had already a look at the new diagram provided by PR #1907: it's great! 
* The only major problem with it may be that the non-unital Ring (Rng) is still in my Mathbox only - I would appreciate if it can be moved to main set.mm (there was no consensus for that until now...). 
* the arrow to Neutral Element should come from Mnd (Monoid), not from SGrp (SemiGroup)
* what about SRing (semirings)? A SRing is a CMnd with a Mnd as multiplicative group (see ~issrg), and a Ring is a SRing (~ringsrg)
* A Rng is an abelian Group (Abel) with a SGrp as multiplicative group (see ~isrng)
* the mulGrp of a Ring is a Mnd, not a Grp 

Regarding one of the three remaining bullet points:

* new graph definitions "as structures": (UHGraph , USHGraph, UPGraph, UMGraph, USPGraph, USGraph, FinUSGraph, ComplGraph, ComplUSGraph): the revised approach for Graph theory is is still under construction (within my Mathbox) - you should wait until the current approach is replaced by the new approach, or make at least clear that there are two approaches at the moment (and provide a diagarm with the currently official definitions UHGrph, etc.).

**Comment by savask** [comment]
I think a small update to the text is needed also, for instance text under fig. 2 says:
>There is currently no structure in set.mm representing a magma, but this does not seem to be an issue.

even though set.mm contains `Mgm` now.

And speaking of non-unital rings, I hope they will be moved to main set.mm (and will supersede unital rings). Rng theory is more regular in a way (ideals of rngs are again rngs and so on), and there are plenty of meaningful theorems for rngs which make little sense for rings (for example, a semisimple artinian rng contains unity). I couldn't find any discussions about that on the metamath group though, so maybe I'm missing some arguments against moving rngs to main.

**Comment by avekens** [comment]
@savask The discussion about Rng was/is in issue #1389

**Comment by savask** [comment]
@avekens Thank you, now I see what's the issue. In that case I hope someone will prove a "metamath 100"-level theorem for rngs to leave no doubts that non-unital rings are useful.
